example/k3d/environment/main.jsonnet

```diff
@@ -51,13 +57,66 @@ local images = {
     }) +
     grafana_agent.withRemoteWrite([{
       url: 'http://cortex.default.svc.cluster.local/api/prom/push',
-    }]),
+    }]) +
+
+    //
+    // Configure Tempo
+    //
+
+    grafana_agent.withTempoConfig({
+      receivers: {
+        jaeger: {
+          protocols: {
+            thrift_http: null,
+            thrift_binary: null,
+            thrift_compact: null,
+            grpc: null,
+          },
+        },
+      },
+    }) + 
+    grafana_agent.withPortsMixin([
+      // Jaeger receiver
+      containerPort.new('thrift-c', 6831) + containerPort.withProtocol('UDP'),
+      containerPort.new('thrift-bin', 6832) + containerPort.withProtocol('UDP'),
+      containerPort.new('thrift-http', 14268) + containerPort.withProtocol('TCP'),
+      containerPort.new('thrift-grpc', 14250) + containerPort.withProtocol('TCP'),
+    ]) +
+    grafana_agent.withTempoPushConfig({
+      endpoint: 'tempo:55680',
+      insecure: true,
+      batch: {
+        timeout: '5s',
+        send_batch_size: 100,
+      },
+    }) + 
+    grafana_agent.withTempoScrapeConfigs([
+      cfg {
+        // We want our cluster and label to remain static for this deployment, so
+        // if they are overwritten by a metric we will change them to the values
+        // set by external_labels.
+        relabel_configs+: [{
+          target_label: 'cluster', replacement: cluster_label,
+        }]
+      }
+      for cfg in grafana_agent.tempoScrapeKubernetes
+    ]),
 
   // Need to run ETCD for agent_cluster
   etcd: etcd.new('default'),
 
   agent_cluster:
     agent_cluster.new('default', 'kube-system') +
+    agent_cluster.withEnvMixin([
+      // Configure the Agent to send spans to the DaemonSet using thrift_compact.
+      { name: 'JAEGER_AGENT_HOST', value: 'grafana-agent.default.svc.cluster.local' },
+      { name: 'JAEGER_AGENT_PORT', value: '6831' },
+
+      // Send every span. You wouldn't want this in prod, but our span
+      // throughout here is low enough that we need a high sampling rate.
+      { name: 'JAEGER_SAMPLER_TYPE', value: 'const' },
+      { name: 'JAEGER_SAMPLER_PARAM', value: '1' },
+    ]) +
```

Comment:
What do you think about using the [jaeger-agent-mixin](https://github.com/grafana/jsonnet-libs/tree/master/jaeger-agent-mixin)? It might not be the ideal :man_shrugging:

Reply:
Good idea, though it really wouldn't work here, since the jaeger-agent-mixin is expecting to modify the container object directly, which is fairly hidden away from the Agent library's API. For a quick and hacky example, I think just setting the environment variables directly like this is probably good enough.